### PR TITLE
code cleanup

### PR DESCRIPTION
**Pull request by dsyme**
* [x] split ast.fs into XmlDoc.fs, AbstractSyntax.fs, AbstractSyntaxOps.fs, ParseHelpers.fs
* [x] regularize pars.fsy
* [x] use `module rec` in AbstractSyntax.fs to give clearer domain model 
* [x] rename `SequencePoint` to `DebugPoint` and use `RequireQualifiedAccess` for simpler names
* [x] split and reformat several really long lines ( > 250 characters)
* [x] remove non-idiomatic indentation in `PrettyNaming.fs`
* [x] put `open` declarations in order in several places
* [x] add regular spacing in `lex.fsl` (it was irregular before)
* [x] some other minor cosmetic cleanup

### CODE REVIEW

File: src/fsharp/FSharp.Core/prim-types-prelude.fs [hunk 1]
```diff
@@ -44,24 +44,42 @@ namespace Microsoft.FSharp.Core
...
-    type ``[,,,,,,,,,,,,,,,,,,,,,,,,,,,,]``<'T> = (# "!0[0 ...,0 ...,0 ...,0 ...,0 ...,0 ...,0 ...,0 ...,0 ...,0 ...,0 ...,0 ...,0 ...,0 ...,0 ...,0 ...,0 ...,0 ...,0 ...,0 ...,0 ...,0 ...,0 ...,0 ...,0 ...,0 ...,0 ...,0 ...,0 ...]" #)
-    type ``[,,,,,,,,,,,,,,,,,,,,,,,,,,,,,]``<'T> = (# "!0[0 ...,0 ...,0 ...,0 ...,0 ...,0 ...,0 ...,0 ...,0 ...,0 ...,0 ...,0 ...,0 ...,0 ...,0 ...,0 ...,0 ...,0 ...,0 ...,0 ...,0 ...,0 ...,0 ...,0 ...,0 ...,0 ...,0 ...,0 ...,0 ...,0 ...]" #)
-    type ``[,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,]``<'T> = (# "!0[0 ...,0 ...,0 ...,0 ...,0 ...,0 ...,0 ...,0 ...,0 ...,0 ...,0 ...,0 ...,0 ...,0 ...,0 ...,0 ...,0 ...,0 ...,0 ...,0 ...,0 ...,0 ...,0 ...,0 ...,0 ...,0 ...,0 ...,0 ...,0 ...,0 ...,0 ...]" #)
-    type ``[,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,]``<'T> = (# "!0[0 ...,0 ...,0 ...,0 ...,0 ...,0 ...,0 ...,0 ...,0 ...,0 ...,0 ...,0 ...,0 ...,0 ...,0 ...,0 ...,0 ...,0 ...,0 ...,0 ...,0 ...,0 ...,0 ...,0 ...,0 ...,0 ...,0 ...,0 ...,0 ...,0 ...,0 ...,0 ...]" #)
+    type ``[,,,,,,,,,,,,,,]``<'T> =
+        (# "!0[0 ...,0 ...,0 ...,0 ...,0 ...,0 ...,0 ...,0 ...,0 ...,0 ...,0 ...,0 ...,0 ...,0 ...,0 ...]" #)
```
Comment (auduchinok) [inline src/fsharp/FSharp.Core/prim-types-prelude.fs]: Maybe add empty lines between types, since these declarations are multiline now?

File: src/fsharp/FSharp.Core/prim-types.fs [hunk 2]
```diff
@@ -135,7 +135,9 @@ namespace Microsoft.FSharp.Core
     type StructuralEqualityAttribute() = 
         inherit System.Attribute()
 
-    [<AttributeUsage (AttributeTargets.Class ||| AttributeTargets.Interface ||| AttributeTargets.Delegate ||| AttributeTargets.Struct ||| AttributeTargets.Enum, AllowMultiple=false)>]  
+    [<AttributeUsage (AttributeTargets.Class ||| AttributeTargets.Interface |||
+                      AttributeTargets.Delegate ||| AttributeTargets.Struct |||
```
Comment (auduchinok) [inline src/fsharp/FSharp.Core/prim-types.fs]: The `AttributeTargets.Delegate` would fit on the previous line (if we're aiming 120 symbols line width) and the whole attribute list could then be placed on two lines instead.

File: src/fsharp/xlf/FSComp.txt.es.xlf [hunk 3]
```diff
@@ -329,7 +329,7 @@
       </trans-unit>
       <trans-unit id="followingPatternMatchClauseHasWrongType">
         <source>All branches of a pattern match expression must return values of the same type as the first branch, which here is '{0}'. This branch returns a value of type '{1}'.</source>
-        <target state="translated">Todas las ramas de una expresión de coincidencia de patrón deben devolver valores del mismo tipo. La primera rama devolvió un valor de tipo "{0}", pero esta rama devolvió un valor de tipo "\{1 \}".</target>
+        <target state="new">All branches of a pattern match expression must return values of the same type as the first branch, which here is '{0}'. This branch returns a value of type '{1}'.</target>
```
Comment (auduchinok) [inline src/fsharp/xlf/FSComp.txt.es.xlf]: Was this change intended?

File: src/fsharp/service/ServiceParseTreeWalk.fs [hunk 4]
```diff
@@ -412,8 +412,8 @@ module public AstTraversal =
                      dive synExpr2 synExpr2.Range traverseSynExpr]
                     |> pick expr
                 | SynExpr.Lazy (synExpr, _range) -> traverseSynExpr synExpr
-                | SynExpr.SequentialOrImplicitYield (_sequencePointInfoForSeq, synExpr, synExpr2, _, _range) 
-                | SynExpr.Sequential (_sequencePointInfoForSeq, _, synExpr, synExpr2, _range) -> 
+                | SynExpr.SequentialOrImplicitYield (_SequencePointInfoForSequential, synExpr, synExpr2, _, _range) 
+                | SynExpr.Sequential (_SequencePointInfoForSequential, _, synExpr, synExpr2, _range) ->
```
Comment (auduchinok) [inline src/fsharp/service/ServiceParseTreeWalk.fs]: It seems a local name is now upper-case, it doesn't look correct style-wise to me.

File: src/fsharp/FSharp.Core/prim-types.fs (hunk 2 above)
Comment (dsyme) [reply src/fsharp/FSharp.Core/prim-types.fs]: When I split I often just split to approx 80 I think

Comment (auduchinok) [reply src/fsharp/FSharp.Core/prim-types.fs]: It's subjective, but I'd suggest using 120 which GitHub/Fantomas default now.

File: src/fsharp/service/ServiceParseTreeWalk.fs (hunk 4 above)
Comment (dsyme) [reply src/fsharp/service/ServiceParseTreeWalk.fs]: good catch thanks, was a case-insensitive global replace

File: src/fsharp/xlf/FSComp.txt.es.xlf (hunk 3 above)
Comment (dsyme) [reply src/fsharp/xlf/FSComp.txt.es.xlf]: no, strange

**Comment by dsyme** [comment]
This is ready to go in once approved

**Comment by cartermp** [comment]
@dsyme could you rename this PR to reflect a more general refactoring/formatting, since it's more than just those two files?

File: src/fsharp/FSharp.Core/prim-types-prelude.fs [hunk 5]
```diff
@@ -31,37 +31,86 @@ namespace Microsoft.FSharp.Core
...
     type ``[,,,,,,,,]``<'T> = (# "!0[0 ...,0 ...,0 ...,0 ...,0 ...,0 ...,0 ...,0 ...,0 ...]" #)
+
     type ``[,,,,,,,,,]``<'T> = (# "!0[0 ...,0 ...,0 ...,0 ...,0 ...,0 ...,0 ...,0 ...,0 ...,0 ...]" #)
+
     type ``[,,,,,,,,,,]``<'T> = (# "!0[0 ...,0 ...,0 ...,0 ...,0 ...,0 ...,0 ...,0 ...,0 ...,0 ...,0 ...]" #)
+
     type ``[,,,,,,,,,,,]``<'T> = (# "!0[0 ...,0 ...,0 ...,0 ...,0 ...,0 ...,0 ...,0 ...,0 ...,0 ...,0 ...,0 ...]" #)
+
     type ``[,,,,,,,,,,,,]``<'T> = (# "!0[0 ...,0 ...,0 ...,0 ...,0 ...,0 ...,0 ...,0 ...,0 ...,0 ...,0 ...,0 ...,0 ...]" #)
+
```
Comment (KevinRansom) [inline src/fsharp/FSharp.Core/prim-types-prelude.fs]: You had to make so I got to see these lines again.  These all make me feel sad.

**Comment by KevinRansom** [comment]
````
         D:\a\1\s\src\fsharp\TypeChecker.fs(15168,46): error FS3174: The union case 'Field' does not have a field named 'idOpt'. [D:\a\1\s\fcs\FSharp.Compiler.Service\FSharp.Compiler.Service.fsproj]
D:\a\1\s\src\fsharp\TypeChecker.fs(15168,46): error FS3174: The union case 'Field' does not have a field named 'idOpt'. [D:\a\1\s\fcs\FSharp.Compiler.Service\FSharp.Compiler.Service.fsproj]
````

File: src/fsharp/FSharp.Core/prim-types-prelude.fs (hunk 5 above)
Comment (dsyme) [reply src/fsharp/FSharp.Core/prim-types-prelude.fs]: :-)

**Comment by dsyme** [comment]
BTW I think that, within the compilerm we should rename "SequencePoint" to "DebugPoint", and only use "SequencePoint" when emitting in the back end.  

This is because the compile already uses "Sequence" and "Seq" confusingly for

* Sequential statements `expr1; expr2`
* Sequence expressions

Plus 
* `DebugPoint` is shorter than `SequencePoint`
* `DebugPoint` is much more self-explanatory

**Comment by auduchinok** [comment]
Splitting files effectively removes the git history. It's practically impossible to use git blame/annotations on previously split files now. :(
Do we really need it?

**Comment by dsyme** [comment]
> Splitting files effectively removes the git history. It's practically impossible to
> use git blame/annotations on previously split files now. :( Do we really need it?

Yes, I understand.  However I don't think we should aim to avoid this because of this limitation in git. Several files in the compiler are way too long and in several mix a number of unrelated things.

In this case, ast.fs becomes much simpler to read if the domain model is in one file (using `module rec`), separate to the other things.  I've taken several beginners through ast.fs recently and this change will greatly improve its readability and allow us to continue to document it, and also help stabilise it as part of the FCS public API.

**Comment by matthid** [comment]
To workaround the history there is https://stackoverflow.com/a/53849651/1269722 but I'm not sure if this survives squashing (but it probably should)

**Comment by dsyme** [comment]
> To workaround the history there is https://stackoverflow.com/a/53849651/1269722 but I'm not sure if this survives squashing (but it probably should)

I'd be very surprised if this survives squashing?  That makes just one commit, with no merges, and that technique is based on merges right?

**Comment by matthid** [comment]
To be honest I have no idea. But I wouldn't be surprised if it would work, given that the internal representation might look differently after those operations (considering how git works with pointers and content objects internally)

**Comment by dsyme** [comment]
I've listed out the various items of code cleanup in the comments

**Comment by dsyme** [comment]
A reminder that anyone who wants to help with code sanitization in the F# compiler codebase can simply run

    fsi tests\scripts\codingConventions.fsx

and a report is printed highlighting some basic coding conventions violations including

* overly long lines
* lack of space after comma
* dangling semicolons
* no space after colon
* space before colon (note: sometimes acceptable)
* internal spacing in lines 
* parenthesized atomic expressions
* too many `cenv.g` (can be reduced by binding `let g = cenv.g` appropriately, makes cleaner code)

We could add many further checks to these, e.g. to achieve naming regularity (`typ` not `ty` for example, also many many other instances of this)

**Comment by dsyme** [comment]
@matthid @auduchinok I decided not to try to apply the technique above, my understanding of the internals of git is not good enough in case there is a problem with it.
